lib/IRGen/ClassLayout.h

```diff
@@ -39,6 +39,59 @@ enum class FieldAccess : uint8_t {
   ConstantIndirect
 };
 
+enum class ClassMetadataFlags {
+  // The below flags indicate various things about the metadata of the
+  // class that might require dynamic initialization or resilient
+  // access patterns:
+
+  // Does the class or any of its superclasses have stored properties that
+  // where dropped due to the Swift language version availability of
+  // their types?
```

Comment:
Nitpick: These should all be doc comments, and the paragraph above should be the comment for the type.